src/components/Autocomplete/Autocomplete.js

```diff
@@ -0,0 +1,382 @@
+                                        ...elem('list', stateAndProps)
+                                    })}
+                                >
+                                    {focused
+                                        ? this.renderSuggestions({
+                                              getItemProps,
+                                              highlightedIndex
+                                          })
+                                        : null}
+                                </List>
+                            </InputWrapper>
+                        </div>
+                    )}
+                </Downshift>
+            </div>
+        );
+    }
+}
+
+Autocomplete.propTypes = {
+    /** array of already selected suggestions */
+    selectedSuggestions: PropTypes.array, // eslint-disable-line react/forbid-prop-types
+    /** getSuggestions(inputValue) => should return an array of objects that will be used to render the suggestions list. */
+    getSuggestions: PropTypes.func.isRequired,
+    /** suggestionToString(suggestion) should return a string to be displayed in the UI. e.g.: suggestion => suggestion.name */
+    suggestionToString: PropTypes.func.isRequired,
+    /** if suggestions are still loading, i.e. display palceholders */
+    isLoading: PropTypes.bool,
+    /** a string or function (to be called with selectedValues) that represents the selected values when the component is blured */
+    selectedPlaceholder: PropTypes.string,
+    /** to be shown in the input field when no value is typed */
+    inputPlaceholder: PropTypes.string.isRequired,
+    /** to be shown when no suggestions are available */
+    noSuggestionsPlaceholder: PropTypes.string.isRequired,
+    /** to be shown as clear button title */
+    clearTitle: PropTypes.string.isRequired,
+    /** onBlur() is called when the component is blurred */
+    onBlur: PropTypes.func,
+    /** onSelectionChange() called when a suggestion is seelcted or removed. Can be used to implement the component as controlled component */
+    onSelectionChange: PropTypes.func.isRequired,
+    /** onInputValueChange(inputValue) called when the input values is changed. Can be used to implement the component as controlled component */
+    onInputValueChange: PropTypes.func,
+    /** reset the selected suggestions array to it's default value */
+    onClearAllSelected: PropTypes.func,
+    /** show Clear button on hover even if there are no selectedSuggestions passed */
+    showClearButton: PropTypes.bool,
+    /** an icon or other node to always be rendered as a first element inside the input box */
+    iconNode: PropTypes.node,
+    /** should this component behaive as a multiselect (e.g. no collapse after selection made) */
+    isMultiselect: PropTypes.bool,
+    /** style the compoent to be prominent */
+    isProminent: PropTypes.bool
+};
+
+Autocomplete.defaultProps = {
+    selectedSuggestions: null,
+    isLoading: false,
```

Comment:
if it handles loading indication, should it also handle request fail indication?

Reply:
Not sure. we have no designs at all for how to do error handling at all. If that comes in pop ups, or within the component... I will leave it out of scope, until we have designs